packages/databricks-vscode/src/bundle/BundleProjectManager.ts

```diff
@@ -0,0 +1,254 @@
+        await this._isBundleProject.refresh();
+
+        // The cached value updates subProjectsAvailabe context.
+        // We have a configurationView that shows "open project" button if the context value is true.
+        await this._subProjects.refresh();
+
+        const isLegacyProject = await this._isLegacyProject.value;
+        if (isLegacyProject) {
+            this.logger.debug(
+                "Detected a legacy project.json, starting automatic migration"
+            );
+            await this.migrateProjectJsonToBundle();
+        }
+    }
+
+    private async configureBundleProject() {
+        if (await this.isBundleProject()) {
+            this.logger.debug(
+                "Detected an existing bundle project, initializing project services"
+            );
+            return this.initProjectServices();
+        } else {
+            this.logger.debug(
+                "No bundle config detected, disposing project services"
+            );
+            await this.disposeProjectServices();
+        }
+    }
+
+    @Mutex.synchronise("projectServicesMutex")
+    private async initProjectServices() {
+        if (this.projectServicesReady) {
+            this.logger.debug("Project services have already been initialized");
+            return;
+        }
+        await this.configModel.init();
+        await this.connectionManager.init();
+        this.projectServicesReady = true;
+    }
+
+    private async disposeProjectServices() {
+        // TODO
+    }
+
+    public async openSubProjects() {
+        const projects = await this._subProjects.value;
+        if (projects.length === 0) {
+            return;
+        }
+        return this.promptToOpenSubProjects(projects);
+    }
+
+    private async promptToOpenSubProjects(
+        projects: {absolute: Uri; relative: Uri}[]
+    ) {
+        type OpenProjectItem = QuickPickItem & {uri?: Uri};
+        const items: OpenProjectItem[] = projects.map((project) => {
+            return {
+                uri: project.absolute,
+                label: project.relative.fsPath,
+                detail: project.absolute.fsPath,
+            };
+        });
+        items.push(
+            {label: "", kind: QuickPickItemKind.Separator},
+            {label: "Choose another folder"}
+        );
+        const options = {
+            title: "We've detected several Databricks projects, select the one you want to open",
+        };
+        const item = await window.showQuickPick<OpenProjectItem>(
+            items,
+            options
+        );
+        if (!item) {
+            return;
+        }
+        await commands.executeCommand("vscode.openFolder", item.uri);
+    }
+
+    private async migrateProjectJsonToBundle() {
+        // TODO
+    }
+
+    public async initNewProject() {
+        const authProvider = await LoginWizard.run(this.cli, this.configModel);
+        if (!authProvider || !(await authProvider.check())) {
+            return;
+        }
```

Comment:
Can we show users a prompt if they want to use the already connected authentication method? In that case, we can use auth provider from connection manager.

Reply:
Currently the command to init new project is only exposed in non-logged-in state. But let me expose it in the command palette to as "Initialise new project using existing auth" and update the logic